### PR TITLE
make null lowercase a missing value

### LINKED ISSUE #16471: Add 'null' in lower case to na_values defaults in pandas.read_csv
Docs to read_csv says that by default 'NULL' in upper case is interpreted as NaN but not in lower case.
It became an issue with the latest yahoo change that affected pandas-datareader, see 
Exception downloading yahoo historical data with  adjust_price=True on  fix-yahoo branch
https://github.com/pydata/pandas-datareader/issues/342

### PR DESCRIPTION
**Pull request by OlegShteynbuk**
xref #16079
closes https://github.com/pandas-dev/pandas/issues/16471

### CODE REVIEW

**Comment by gfyoung** [comment]
1) You need to add it to `parsers.pyx`
2) There's a test (also in `na_values.py`) that checks whether each NaN value you specified gets interpreted as such.  You should add `null` to that test.

**Comment by OlegShteynbuk** [comment]
Thanks,
1. will add it to parsers.pyx - _NA_VALUES line 279
2. na_values.py - added to the test, method test_default_na_values()  , should be in the pull request.BTW was not able to run pytest, maybe because the test class name doesn't start with the word test;wrote my own simple test with the data that triggered this issue, it passed.
 

    On Monday, May 29, 2017 4:15 PM, gfyoung <notifications@github.com> wrote:
 

    
   - You need to add it to parsers.pyx
   - There's a test (also in na_values.py) that checks whether each NaN value you specified gets interpreted as such. You should add null to that test.

**Comment by gfyoung** [comment]
@OlegShteynbuk : Yeah, you need to run `pytest test_parsers.py` instead (and yeah, that was the test I was referring to.  Good catch :smile:)

**Comment by OlegShteynbuk** [comment]
run pytest twice, second time with the class name, some tests skipped, all others passed, should be ok, right?
 pytest pandas/tests/io/parser/test_parsers.
725 passed, 13 skipped in 76.04 seconds

 pytest pandas/tests/io/parser/test_parsers.py::TestPythonParser
235 passed, 5 skipped in 30.06 seconds 
 

    On Monday, May 29, 2017 4:29 PM, gfyoung <notifications@github.com> wrote:
 

 @OlegShteynbuk : Yeah, you need to run pytest test_parsers.py instead (and yeah, that was the test I was referring to. Good catch 😄 )—

**Comment by gfyoung** [comment]
@OlegShteynbuk : Nope, not an issue.

**Comment by OlegShteynbuk** [comment]
added it to parsers.pyx thanks to gfyoung

**Comment by gfyoung** [comment]
@OlegShteynbuk : See the checklist in the PR you submitted?  Just run through those and make sure you can check all of them off.  Then it should be good to merge!

**Comment by OlegShteynbuk** [comment]
@gfyoung looks ok, not sure if the last test change to parsers.pyx change is included, or need a new pull request.

**Comment by gfyoung** [comment]
I'm not sure you fully understood what I was asking you to do.  The code changes are fine, but there are several other things you need to do (i.e whatsnew) before this will be merged.  That's why we have the checkboxes in the PR you submitted

**Comment by OlegShteynbuk** [comment]
where are the checkboxes located, can see checks for tests they all fine, i can write whatsnew but where is it?

**Comment by gfyoung** [comment]
If you look at the web version of GitHub, the very first text box under the PR title has the checkboxes.  Also, the whatsnew can be found under `doc/source/whatsnew` (probably want 0.21.0 as this is technically an API change)

**Comment by OlegShteynbuk** [comment]
there are checkboxes on the top, first comment, there is whatsnew checkbox don't see any links, need to add whatsnew somewhere

**Comment by gfyoung** [comment]
> there are checkboxes on the top, first comment

Correct.  Make sure you've satisfied everything listed on those checkboxes.

> need to add whatsnew somewhere

Yes, I gave you the file-path that you should follow in your local repository: `pandas/doc/source/whatsnew`

**Comment by OlegShteynbuk** [comment]
probably need a new .txt file under pandas/doc/source/whatsnew

**Comment by gfyoung** [comment]
No, you shouldn't.  You should add it to the 0.21 version.

**Comment by OlegShteynbuk** [comment]
ok, will add 0.21

**Comment by OlegShteynbuk** [comment]
added whatsnew, commited to my branch

**Comment by OlegShteynbuk** [comment]
all other checkboxes looks fine, will be away for about an hour, do i need to do anything  else before leaving?

**Comment by OlegShteynbuk** [comment]
- The ``na_values`` argument for :func:`read_csv` function now has 'null' (in lower case) as a default value that is interpreted as NaN: (:issue:`16471`)

**Comment by OlegShteynbuk** [comment]
probably need to modify na_values in io.rst file too as it goes into docs
added 'null' to io.rst line 226,  looks like the one that is going into read_csv docs, on my branch; 
there is also a heading NA Values  (line 1023) describing default values, not sure about this one, most likely need to be modified too.

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -38,6 +38,7 @@ Other Enhancements
 - :func:`read_feather` has gained the ``nthreads`` parameter for multi-threaded operations (:issue:`16359`)
 - :func:`DataFrame.clip()` and :func: `Series.cip()` have gained an inplace argument. (:issue: `15388`)
 - :func:`crosstab` has gained a ``margins_name`` parameter to define the name of the row / column that will contain the totals when margins=True. (:issue:`15972`)
+- The ``na_values`` argument for :func:`read_csv` function now has 'null' (in lower case) as a default value that is interpreted as NaN: (:issue:`16471`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: ``:func:`read_csv`` has gained ``'null'`` as an additional default missing value.

``na_values`` is not involved here, rather the default has changed.

**Comment by OlegShteynbuk** [comment]
maybe to add some clarification:
:func:`read_csv has gained 'null' as an additional default missing value,  see in docs description of ``na_values`` argument for `read_csv` function 

This text was in reference to docs - http://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html    that describes all arguments to read_csv.

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (OlegShteynbuk) [inline doc/source/whatsnew/v0.21.0.txt]: maybe to add some clarification:
:func:read_csv has gained 'null' as an additional default missing value, see in docs description of ``na_values`` argument forread_csv` function

This text was in reference to docs - http://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html that describes all arguments to read_csv.

Comment (OlegShteynbuk) [inline doc/source/whatsnew/v0.21.0.txt]: read_csv is not the only function that has na_values argument, maybe change to:

The default NaN recognized values has gained 'null' as an additional default value,  see in docs description of na_values argument for any of the following functions: `read_csv`, `read_table`, `read_excel`, or `read_fwf`

Comment (OlegShteynbuk) [inline doc/source/whatsnew/v0.21.0.txt]: changed;  yes -  ``na_values`` is just a way to add additional strings to those default values for one function call, thanks for the correction.

**Comment by jreback** [comment]
thanks @OlegShteynbuk

**Comment by jreback** [comment]
@gfyoung do we have an issue about consolidating the duplicate definitions of NA values (in parsers & IO), maybe just remove from parsers and leave in IO is best. if not, can you create one?

**Comment by OlegShteynbuk** [comment]
consolidated the duplicate definitions of NA values (in parsers & IO), removed from parsers and left in IO, assert removed too;  probably need to create a different issue.
https://github.com/pandas-dev/pandas/pull/16589

**Comment by OlegShteynbuk** [comment]
consolidated NA values in parsers.pyx

**Pull request merged**